test/prism_regression/flip_flop.rb

```diff
@@ -0,0 +1,26 @@
+# typed: false
+
+if "flip".."flop" # 2 dots
+  #^^^^^^^^^^^^^^ error: Unsupported node type `IFlipflop`
+  "body 1"
+end
+
+if "flip"..."flop" # 3 dots
+  #^^^^^^^^^^^^^^^ error: Unsupported node type `EFlipflop`
+  "body 2"
+end
+
+"body 3" if "flip".."flop" # In a modifier
+#           ^^^^^^^^^^^^^^ error: Unsupported node type `IFlipflop`
+
+unless "flip".."flop"
+  #    ^^^^^^^^^^^^^^ error: Unsupported node type `IFlipflop`
+  "body 4"
+end
+
+"body 5" unless "flip".."flop" # In a modifier
+#               ^^^^^^^^^^^^^^ error: Unsupported node type `IFlipflop`
+
+# In a boolean expression
+b = !("flip".."flop")
```

Comment:
This is somehow _even more wild_ than it just being special-cased to `if` and `unless`.

It happens with `!`, but not `&&` or `||`.

```ruby
Prism.parse "(foo..bar) || (flip..flop)"
```

<details><summary>Output</summary>

```
#<Prism::ParseResult:0x000000012699e6f8 @value=@ ProgramNode (location: (1,0)-(1,26))
├── flags: ∅
├── locals: []
└── statements:
    @ StatementsNode (location: (1,0)-(1,26))
    ├── flags: ∅
    └── body: (length: 1)
        └── @ OrNode (location: (1,0)-(1,26))
            ├── flags: newline
            ├── left:
            │   @ ParenthesesNode (location: (1,0)-(1,10))
            │   ├── flags: ∅
            │   ├── body:
            │   │   @ StatementsNode (location: (1,1)-(1,9))
            │   │   ├── flags: ∅
            │   │   └── body: (length: 1)
            │   │       └── @ RangeNode (location: (1,1)-(1,9))
            │   │           ├── flags: newline
            │   │           ├── left:
            │   │           │   @ CallNode (location: (1,1)-(1,4))
            │   │           │   ├── flags: variable_call, ignore_visibility
            │   │           │   ├── receiver: ∅
            │   │           │   ├── call_operator_loc: ∅
            │   │           │   ├── name: :foo
            │   │           │   ├── message_loc: (1,1)-(1,4) = "foo"
            │   │           │   ├── opening_loc: ∅
            │   │           │   ├── arguments: ∅
            │   │           │   ├── closing_loc: ∅
            │   │           │   └── block: ∅
            │   │           ├── right:
            │   │           │   @ CallNode (location: (1,6)-(1,9))
            │   │           │   ├── flags: variable_call, ignore_visibility
            │   │           │   ├── receiver: ∅
            │   │           │   ├── call_operator_loc: ∅
            │   │           │   ├── name: :bar
            │   │           │   ├── message_loc: (1,6)-(1,9) = "bar"
            │   │           │   ├── opening_loc: ∅
            │   │           │   ├── arguments: ∅
            │   │           │   ├── closing_loc: ∅
            │   │           │   └── block: ∅
            │   │           └── operator_loc: (1,4)-(1,6) = ".."
            │   ├── opening_loc: (1,0)-(1,1) = "("
            │   └── closing_loc: (1,9)-(1,10) = ")"
            ├── right:
            │   @ ParenthesesNode (location: (1,14)-(1,26))
            │   ├── flags: ∅
            │   ├── body:
            │   │   @ StatementsNode (location: (1,15)-(1,25))
            │   │   ├── flags: ∅
            │   │   └── body: (length: 1)
            │   │       └── @ RangeNode (location: (1,15)-(1,25))
            │   │           ├── flags: newline
            │   │           ├── left:
            │   │           │   @ CallNode (location: (1,15)-(1,19))
            │   │           │   ├── flags: variable_call, ignore_visibility
            │   │           │   ├── receiver: ∅
            │   │           │   ├── call_operator_loc: ∅
            │   │           │   ├── name: :flip
            │   │           │   ├── message_loc: (1,15)-(1,19) = "flip"
            │   │           │   ├── opening_loc: ∅
            │   │           │   ├── arguments: ∅
            │   │           │   ├── closing_loc: ∅
            │   │           │   └── block: ∅
            │   │           ├── right:
            │   │           │   @ CallNode (location: (1,21)-(1,25))
            │   │           │   ├── flags: variable_call, ignore_visibility
            │   │           │   ├── receiver: ∅
            │   │           │   ├── call_operator_loc: ∅
            │   │           │   ├── name: :flop
            │   │           │   ├── message_loc: (1,21)-(1,25) = "flop"
            │   │           │   ├── opening_loc: ∅
            │   │           │   ├── arguments: ∅
            │   │           │   ├── closing_loc: ∅
            │   │           │   └── block: ∅
            │   │           └── operator_loc: (1,19)-(1,21) = ".."
            │   ├── opening_loc: (1,14)-(1,15) = "("
            │   └── closing_loc: (1,25)-(1,26) = ")"
            └── operator_loc: (1,11)-(1,13) = "||"
```

</details>